viewer/src/public/RevealManagerBase.ts

```diff
@@ -72,21 +114,191 @@ export class RevealManagerBase<TModelIdentifier> implements RenderManager {
+    return this._cadManager.renderMode;
+  }
+
+  public set renderMode(renderMode: RenderMode) {
+    this._cadManager.renderMode = renderMode;
+  }
+
   public set clippingPlanes(clippingPlanes: THREE.Plane[]) {
-    this._materialManager.clippingPlanes = clippingPlanes;
     this._cadManager.clippingPlanes = clippingPlanes;
   }
 
   public get clippingPlanes(): THREE.Plane[] {
-    return this._materialManager.clippingPlanes;
+    return this._cadManager.clippingPlanes;
   }
 
   public set clipIntersection(intersection: boolean) {
-    this._materialManager.clipIntersection = intersection;
     this._cadManager.clipIntersection = intersection;
   }
 
   public get clipIntersection() {
-    return this._materialManager.clipIntersection;
+    return this._cadManager.clipIntersection;
+  }
+
+  public on(event: 'loadingStateChanged', listener: LoadingStateChangeListener): void;
+  public on(event: 'nodeIdToTreeIndexMapLoaded', listener: SectorNodeIdToTreeIndexMapLoadedListener): void;
+  public on(
+    event: 'loadingStateChanged' | 'nodeIdToTreeIndexMapLoaded',
+    listener: LoadingStateChangeListener | SectorNodeIdToTreeIndexMapLoadedListener
+  ): void {
+    switch (event) {
+      case 'loadingStateChanged':
+        this.eventListeners.loadingStateChanged.push(listener as LoadingStateChangeListener);
+        break;
+
+      case 'nodeIdToTreeIndexMapLoaded':
+        this.eventListeners.sectorNodeIdToTreeIndexMapLoaded.push(listener as SectorNodeIdToTreeIndexMapLoadedListener);
+        break;
+
+      default:
+        throw new Error(`Unsupported event '${event}'`);
+    }
+  }
+
+  public off(event: 'loadingStateChanged', listener: LoadingStateChangeListener): void;
+  public off(event: 'nodeIdToTreeIndexMapLoaded', listener: SectorNodeIdToTreeIndexMapLoadedListener): void;
+  public off(
+    event: 'loadingStateChanged' | 'nodeIdToTreeIndexMapLoaded',
+    listener: LoadingStateChangeListener | SectorNodeIdToTreeIndexMapLoadedListener
+  ): void {
+    switch (event) {
+      case 'loadingStateChanged':
+        this.eventListeners.loadingStateChanged.filter(x => x !== listener);
+        break;
+
+      case 'nodeIdToTreeIndexMapLoaded':
+        this.eventListeners.sectorNodeIdToTreeIndexMapLoaded.filter(x => x !== listener);
+        break;
+
+      default:
+        throw new Error(`Unsupported event '${event}'`);
+    }
+  }
+
+  protected loadModel(
+    type: 'cad',
+    modelIdentifier: TModelIdentifier,
+    nodeApperanceProvider?: NodeAppearanceProvider
+  ): Promise<CadNode>;
+  protected loadModel(
+    type: 'pointcloud',
+    modelIdentifier: TModelIdentifier
+  ): Promise<[PotreeGroupWrapper, PotreeNodeWrapper]>;
+  protected async loadModel(
+    type: 'cad' | 'pointcloud',
+    modelIdentifier: TModelIdentifier,
+    nodeApperanceProvider?: NodeAppearanceProvider
+  ): Promise<[PotreeGroupWrapper, PotreeNodeWrapper] | CadNode> {
+    trackAddModel(
+      {
+        moduleName: 'RevealManager',
```

Comment:
oh, that's lying now. I suggest to use RevealMangerBase and `loadModel` as method name.

Reply:
Yep, change to `moduleName: 'RevealManagerBase'` and `methodName: 'loadModel'` not sure what to do with the `options` parameter here, prbably leave it as is?

Reply:
I think maybe we should detach the events and modules from code, such that the log is `CAD Model::load` instead of `RevealManagerBase::loadModel` to allow code to change without worrying about being able to compare data in MixPanel historically.

Reply:
(Fixed as commented now)